### ISSUE TITLE
No module named 'pkg_resources' with pex 1.6

### ISSUE THREAD

**Issue by thanosexcite** [pantsbuild/pex#631]
When executing a pex that has been created with pex 1.6 it returns the error bellow. If the executable is created with previous versions of pex (like 1.5.3), it runs with no errors.

If I add setuptools in my requirements.txt and build the executable with pex 1.6, it also works without errors. But I don't think having setuptools defined in my requirements.txt is a good thing.

This issue might be related to #624.

```
$ ./foo --help
Traceback (most recent call last):
  File "/home/thanos/foo/.bootstrap/pex/pex.py", line 352, in execute
  File "/home/thanos/foo/.bootstrap/pex/pex.py", line 284, in _wrap_coverage
  File "/home/thanos/foo/.bootstrap/pex/pex.py", line 315, in _wrap_profiling
  File "/home/thanos/foo/.bootstrap/pex/pex.py", line 397, in _execute
  File "/home/thanos/foo/.bootstrap/pex/pex.py", line 495, in execute_entry
  File "/home/thanos/foo/.bootstrap/pex/pex.py", line 509, in execute_pkg_resources
  File "/home/thanos/foo/.bootstrap/pex/vendor/_vendored/setuptools/pkg_resources/__init__.py", line 2352, in resolve
  File "/home/thanos/.pex/install/gunicorn-19.9.0-py2.py3-none-any.whl.fe7cf305dceca558a95c349457530577a4205dd9/gunicorn-19.9.0-py2.py3-none-any.whl/gunicorn/app/wsgiapp.py", line 9, in <module>
    from gunicorn.app.base import Application
  File "/home/thanos/.pex/install/gunicorn-19.9.0-py2.py3-none-any.whl.fe7cf305dceca558a95c349457530577a4205dd9/gunicorn-19.9.0-py2.py3-none-any.whl/gunicorn/app/base.py", line 12, in <module>
    from gunicorn import util
  File "/home/thanos/.pex/install/gunicorn-19.9.0-py2.py3-none-any.whl.fe7cf305dceca558a95c349457530577a4205dd9/gunicorn-19.9.0-py2.py3-none-any.whl/gunicorn/util.py", line 12, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```
The executables are created in a Python 3.6.7 venv using the following:

```
$ pex -vv --disable-cache --python-shebang="/usr/bin/env python3.6" --output-file=foo --requirement=requirements.txt -c gunicorn .
```

**Comment by jsirois** [comment]
> But I don't think having setuptools defined in my requirements.txt is a good thing.

I agree, but it appears gunicorn has a bug in its (missing) [`install_requires`](https://github.com/benoitc/gunicorn/blob/ee7af1247b8cc289af7b4bcfd2ff93ecfaf8c9df/setup.py#L75-L104): it fails to declare its [real dependency on `setuptools`](https://github.com/benoitc/gunicorn/blob/ee7af1247b8cc289af7b4bcfd2ff93ecfaf8c9df/gunicorn/util.py#L24). In the past, pex was leaking its copy of the `pkg_resources` package from `setuptools`; and with 1.6.0 we've plugged the leak.

I'll mark this as answered but close a little later after I've filed an issue / sent a PR to gunicorn unless you beat me to it.
